multierror/multierror.go

```diff
@@ -0,0 +1,28 @@
+package multierror
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Error []error
+
+func (me Error) Error() string {
+	if me == nil {
+		return ""
+	}
+
+	strs := make([]string, len(me))
+	for i, err := range me {
+		strs[i] = fmt.Sprintf("[%d] %v", i, err)
+	}
+	return strings.Join(strs, " ")
+}
+
+func (me Error) ErrorOrNil() error {
```

Comment:
Oh, I see, a function to avoid cases where explicitly turning it into the interface would be required. What about `AsError()`